pkg/schema/v1/contracts.go

```diff
@@ -10,24 +11,32 @@ import (
 	"reflect"
 )
 
+// Private type to prevent collisions with other context keys
+type contextKey string
+
+// clusterContextKey is the key for Cluster values in contexts.
+var clusterContextKey = contextKey("cluster")
```

Comment:
Please rename `clusterContextKey` to `clusterUuidContextKey` and use `contextKey("cluster_uuid")`.